R/dbQueries-Results.R

```diff
@@ -40,13 +40,12 @@ db_send_query <- function(conn, statement, ...) {
 
   # TDOO: if we are in expect_sql, then we should emit an error (or warning?) with SQL here
   if (is_expecting()) {
-    stop(
-      "Fixture: ",
-      make_path(conn@path, get_type(statement), hash(statement)),
-      "\n",
-      clean_statement(statement),
-      call. = FALSE
+    error_msg <- glue::glue(
+      "Fixture: {make_path(conn@path, get_type(statement), hash(statement))}
+      {clean_statement(statement)}"
```

Comment:
What does this look like when it occurs? 

I know there are small differences in how newlines are handled (and it might even depend on if `cli` is installed or not 🙈 ) Would you mind trying to trigger it with and without CLI in your env to see what it looks like and paste that here?

Reply:
Reprex:

```r
devtools::load_all()

with_mock_db({
  conn <- dbConnect(RSQLite::SQLite(), dbname = "not_a_db")
})

statement <- "SELECT * FROM dbo.iris"

error_msg <- glue::glue(
  "Fixture: {make_path(conn@path, get_type(statement), hash(statement))}
      {clean_statement(statement)}"
)

error_msg

rlang::abort(error_msg)
```

Terminal:

![image](https://github.com/ropensci/dittodb/assets/23486577/a97cca78-321b-41e4-a7c7-bb58a0d2a3eb)

Question, what do you mean by without CLI? I removed the {cli} package and ran the `rlang::abort` and it printed the same. I think I am confused :)

Reply:
Hmmm that's odd. I just tried it locally (thanks for the reprex, that made it super easy!), and the content / spacing is the same, but I do see a (tiny) difference without the cli package:

with cli
<img width="434" alt="image" src="https://github.com/ropensci/dittodb/assets/700357/7ff6d6ea-bc86-4834-858c-9126dadb78d5">


without cli
<img width="457" alt="image" src="https://github.com/ropensci/dittodb/assets/700357/84f25e12-b503-459b-a62f-5f4a1756a9dd">

The difference is very subtle here (just the coloring), but that's what I wanted to confirm. I looked at the docs for `rlang::abort` and saw that it will use cli to format the messages (see `use_cli_format ` [under arguments](https://rlang.r-lib.org/reference/abort.html#arguments)), and when I clicked through to [the page on formatting](https://rlang.r-lib.org/reference/topic-condition-formatting.html) there was some discussion about changes in format and whitespace — though those might be limited to only when using `cli::cli_abort` instead of `rlang::abort formatted with cli (I will be honest, I couldn't totally tell from this page when I read it!)

There is another thing on that page that's a little worrying about user input, but that also looks exclusive to `cli::cli_abort` and not `rlang::abort` which has been _formatted with_ cli:

```
> statement <- "SELECT * FROM dbo.iris"
> 
> user_input <- "{base::stop('Wrong message.', call. = FALSE)}"
> 
> error_msg <- glue::glue(
+   "Fixture: ",
+   "{make_path('not_a_db', get_type(statement), hash(statement))}\n",
+   "{clean_statement(statement)}\n",
+   "{user_input}"
+ )
> 
> rlang::abort(error_msg)
Error:
! Fixture: not_a_db/SELECT-eb40ac.R
SELECT * FROM dbo.iris
{base::stop('Wrong message.', call. = FALSE)}
Run `rlang::last_trace()` to see where the error occurred.
```

So I think this is good — though what do you think about:

```
    error_msg <- glue::glue(
      "Fixture: {make_path(conn@path, get_type(statement), hash(statement))}\n",
      "{clean_statement(statement)}"
    )
```

It's mostly a personal preference, but I prefer explicitly escaped new lines as opposed to having it in the quote and overflowing.

Reply:
This also could be incorporated into the `abort` call, though this is starting to get big enough a separate variable is ok. What do you think?